node-packages/commons/src/api.ts

```diff
@@ -919,13 +919,29 @@ export const getDeployTargetConfigsForProject = (project: number): Promise<any>
     }
 `);
 
-export const getOpenShiftInfoForEnvironment = (environment: number): Promise<any> =>
+interface GetOpenShiftInfoForEnvironmentEnvironmentResult {
+  id: number
+  name: string
+  openshiftProjectPattern: string
+  branches: string
+  openshift: DeployTargetMinimalFragment
+  project: Pick<Project, 'sharedBaasBucket' | 'buildImage'> & {
+    envVariables: Pick<EnvKeyValue, 'name' | 'scope' | 'value'>[]
+  }
+}
+
+export interface GetOpenShiftInfoForEnvironmentResult {
+  environment: GetOpenShiftInfoForEnvironmentEnvironmentResult
+}
+
+export const getOpenShiftInfoForEnvironment = (environment: number): Promise<GetOpenShiftInfoForEnvironmentResult> =>
   graphqlapi.query(`
     {
       environment:environmentById(id: ${environment}){
         id
         name
         openshiftProjectPattern
+        branches
```

Comment:
Branches doesn't exist on the environment, same for https://github.com/uselagoon/lagoon/pull/3856/files#diff-8093d432d1ad2f3dfdc094ccd80cdb80d02c03920d0670f412e88d72801bc61aR926

Reply:
Thanks, I figured out that `branches` is never used for that promote path, which  is why it never caused a problem before, and just deleted it. Attempting to strongly type `deployTarget` to handle this better is another huge nightmare, so avoiding it.